api/v1alpha1/s3userclaim_webhook.go

```diff
@@ -224,3 +238,24 @@ func validateAgainstClusterQuota(ctx context.Context, suc *S3UserClaim) error {
 
 	return nil
 }
+
+func validateSecrets(secretNames []string, namespace string, allErrs field.ErrorList) field.ErrorList {
+	ctx, cancel := context.WithTimeout(context.Background(), ValidationTimeout)
+	defer cancel()
+	for _, secretName := range secretNames {
+		existingSecret := &v1.Secret{}
+		switch err := runtimeClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: secretName}, existingSecret); {
+		case apierrors.IsNotFound(err):
+			continue
+		case err != nil:
+			allErrs = append(allErrs, field.InternalError(field.NewPath("spec").Child(secretName), err))
```

Comment:
Are you sure about `.child(secretName)`?

Reply:
@coderabbitai What do you think? Any suggestion?

Reply:
Since being adminSecret or readonlySecret is not distinguishable in the function, what error do you suggest appending?

Reply:
@coderabbitai Since being adminSecret or readonlySecret is not distinguishable in the function, what error do you suggest appending?

Reply:
@coderabbitai I've revised it in the new commit. Check it please.

Reply:
@coderabbitai Sure. Here is the code snippet of the `validateSecrets` function:

```go
func validateSecrets(secretNames []string, namespace string, allErrs field.ErrorList) field.ErrorList {
	ctx, cancel := context.WithTimeout(context.Background(), ValidationTimeout)
	defer cancel()
	for _, secretName := range secretNames {
		existingSecret := &v1.Secret{}
		switch err := runtimeClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: secretName}, existingSecret); {
		case apierrors.IsNotFound(err):
			continue
		case err != nil:
			allErrs = append(allErrs,
				field.InternalError(field.NewPath("spec"), fmt.Errorf("error with getting secret '%s': %w", secretName, err)))
		default:
			allErrs = append(allErrs,
				field.Forbidden(field.NewPath("spec"), fmt.Sprintf("secret %s exists", secretName)))
		}
	}
	return allErrs

}

```